dm/master/server.go

```diff
@@ -151,14 +166,30 @@ func (s *Server) Start(ctx context.Context) (err error) {
 		return
 	}
 
+	// create an etcd client used in the whole server instance.
+	// NOTE: we only use the local member's address now, but we can use all endpoints of the cluster if needed.
+	s.etcdClient, err = etcdutil.CreateClient([]string{s.cfg.MasterAddr})
+	if err != nil {
+		return
+	}
+
 	s.closed.Set(false) // the server started now.
 
+	// start leader election
+	s.election = election.NewElection(ctx, s.etcdClient, electionTTL, electionKey, s.cfg.Name)
```

Comment:
do we need to handle `election`'s error from `election.ErrorNotify()`

Reply:
I update `Election` to let it more like which one in TiDB's owner pkg in https://github.com/pingcap/dm/pull/367/commits/80ae3d23a14157b78fdc4b7ac2a16e424ad18fef and https://github.com/pingcap/dm/pull/367/commits/5c4c8204aeb8eca110439574ea63596c6901752b.

1. Detect any potential error (for unrecoverable error) in `NewElection`, and do not start `Server` if an error occurred.
2. Retry for errors (for recoverable error) until the context is done after entering the campaign loop.
3. receive errors from `election.ErrorNotify()` in _server/election.go_, but do no meaningful things now.